core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6823,6 +6823,48 @@ private void checkSemiJoinRuleOnAntiJoin(RelOptRule rule) {
         .checkUnchanged();
   }
 
+  /** Test case for CALCITE-5683 for two level nested decorrelate with standard program
+   * failing during the decorrelation phase. */
+  @Test void testTwoLevelDecorrelate() {
+    final String sql = "SELECT d1.name, d1.deptno + "
+        + " ( SELECT e1.empno "
+        + " FROM emp e1 "
+        + " WHERE d1.deptno = e1.deptno and "
+        + "       e1.sal = (SELECT max(sal) "
+        + "                 FROM emp e2 "
+        + "                 WHERE   e1.sal = e2.sal and"
+        + "                         e1.deptno = e2.deptno and"
+        + "                         d1.deptno < e2.deptno))"
+        + " FROM dept d1";
+
+    sql(sql)
+        .withExpand(false)
+        .withLateDecorrelate(true)
+        .withTrim(true)
+        .withRule()
+        .checkUnchanged();
```

Comment:
This seems weird that you are asserting the plan is not changed, but the content in `RelOptRulesTest.xml` does contain `planAfter` section, and the tests still pass.

Reply:
Yeah, please let me know if it makes sense to remove these tests (as they actually are redundant) as sub-query.iq is having these test cases and are reporting an error without the fix.

Reply:
I think it's worth to keep these tests, my question is about why `checkUnchanged` test contains `planAfter`, I'll check this and give my further comment.

Reply:
Now I know the reason, and I've logged CALCITE-5727 for further work. For this case, you can delete the `planAfter` section directly in `RelOptRulesTest.xml`, that does not affect the test, and there is no need to add this section.

Reply:
OK, thank you for working on it.